nix/process-compose/default.nix

```diff
@@ -50,5 +56,26 @@ in
         exec process-compose -p ${toString config.port} "$@"
       '';
     };
+  config.outputs.testPackage =
+    let
+      mergedYaml = pkgs.runCommand "${name}-merged.yaml" { buildInputs = [ pkgs.yq-go ]; } ''
+        yq '. *= load("${config.outputs.settingsYamlOverlay}")' ${config.outputs.settingsYaml} > $out
+      '';
+    in
+    pkgs.writeShellApplication {
+      inherit name;
+      runtimeInputs = [ config.package ];
+      text = ''
+        ${if config.debug then "cat ${mergedYaml}" else ""}
+        export PC_CONFIG_FILES=${mergedYaml}
```

Comment:
This can take multiple YAML files, AFAIU

https://github.com/F1bonacc1/process-compose/blob/2da4073b60ddb2b4008eef884295ed3ced0c1141/src/cmd/root.go#L74